src/javascript/_common/os_detect.js

```diff
@@ -0,0 +1,23 @@
+const OSDetect = () => {
+    const Systems = {
+        macos  : ['Mac68K', 'MacIntel', 'MacPPC'],
+        linux  : ['HP-UX', 'Linux i686', 'Linux armv7l', ''],
+        ios    : ['iPhone', 'iPod', 'iPad'],
+        android: ['Android'],
+        windows: ['Win16', 'Win32', 'WinCE'],
+    };
+    if (typeof navigator !== 'undefined') {
```

Comment:
not sure if it's better to change this to `if (navigator && navigator.platform)`

Reply:
or perhaps `getPropertyValue(navigator, 'platform')`

Reply:
Navigator support is so good that we can almost remove this. But, if it is not defined, using it in any other context than `typeof` would throw a reference error. 

Maybe the safer choice would be `type of navigator !== 'undefined' && navigator.hasOwnProperty('platform');`

Reply:
if navigator is undefined the `if (navigator` part will return false and it won't check `&& navigator.platform)` part at all, but it's up to you

Reply:
<img width="330" alt="screen shot 2018-10-03 at 12 01 43" src="https://user-images.githubusercontent.com/188798/46399318-35ba5280-c704-11e8-9beb-b7ccba0ab4f6.png">

Reply:
I was thinking about a declared variable with undefined value :) sorry about that, you're way is fine.